src/arch/arm/32/head.S

```diff
@@ -160,10 +160,15 @@ BEGIN_FUNC(_start)
 
     /* Put the DTB address back on the new stack for init_kernel. */
     push {r7, r8}
-    /* Call bootstrapping implemented in C */
-    blx init_kernel
 
-    /* Restore the initial thread */
-    b restore_user_context
+    /* jump to init_kernel() with parameters:
+     * r0: user image physical start addres
+     * r1: user image physical end address
+     * r2: physical/virtual offset
+     * r3: user image virtual entry address
+     * sp[0]: DTB physical address (0 if there is none)
+     * sp[-1]: DTB size (0 if there is none)
```

Comment:
If this is supposed to be C array syntax, shouldn't the index here be 1 as the stack is decremented on in the push (stmdb sp!) instruction?

Reply:
indeed, that would be more clear. I was wondering what is the best way to describe this.